api/v1alpha1/hostedcluster_types.go

```diff
@@ -114,6 +114,9 @@ const (
 	// removed when deleting the corresponding HostedCluster. If set to "true", resources created on the cloud provider during the life
 	// of the cluster will be removed, including image registry storage, ingress dns records, load balancers, and persistent storage.
 	CleanupCloudResourcesAnnotation = "hypershift.openshift.io/cleanup-cloud-resources"
+
+	// HostedClusterAnnotation is an annotation that indicates which HostedCluster an object corresponds to by namespace/name
```

Comment:
can you remind me why was this moved here? seems we are only consumers ourselves

Reply:
I had asked whether we should move it to the API if it is being used in the NodePool controller. I can move it back to hostedcluster_controller.go and just import the hostedcluster package in nodepool_controller.go if you prefer that over adding it to the API.

Reply:
@enxebre, I undid this change. I don't think there is any point in moving this here, as we are already importing the hostedcluster package in nodepool_controller.go. It keeps the PR much cleaner this way.